sale_order_type/models/sale_order.py

```diff
@@ -1,56 +1,55 @@
 # -*- encoding: utf-8 -*-
 ##############################################################################
-# For copyright and license notices, see __openerp__.py file in root directory
+# For copyright and license notices, see __manifest__.py file in root directory
 ##############################################################################
 
-from openerp import api, fields, models
+from odoo import api, fields, models
 
 
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
     def _get_order_type(self):
-        return self.env['sale.order.type'].search([])[:1]
+        return self.env['sale.order.type'].search([], limit=1)
 
     type_id = fields.Many2one(
         comodel_name='sale.order.type', string='Type', default=_get_order_type)
 
     @api.multi
-    def onchange_partner_id(self, part):
-        res = super(SaleOrder, self).onchange_partner_id(part)
-        if part:
-            partner = self.env['res.partner'].browse(part)
-            res['value'].update({
-                'type_id': partner.sale_type.id or self._get_order_type().id,
-            })
-        return res
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        super(SaleOrder, self).onchange_partner_id()
+        if self.partner_id:
+            self.type_id = self.partner_id.sale_type or self._get_order_type()
 
-    @api.one
+    @api.multi
     @api.onchange('type_id')
     def onchange_type_id(self):
-        self.warehouse_id = self.type_id.warehouse_id
-        self.picking_policy = self.type_id.picking_policy
-        self.order_policy = self.type_id.order_policy
+        for order in self:
+            if order.type_id.warehouse_id:
+                order.warehouse_id = order.type_id.warehouse_id
+            if order.type_id.picking_policy:
+                order.picking_policy = order.type_id.picking_policy
+            if order.type_id.payment_term_id:
+                order.payment_term_id = order.type_id.payment_term_id.id
+            if order.type_id.pricelist_id:
+                order.pricelist_id = order.type_id.pricelist_id.id
+            if order.type_id.incoterm_id:
+                order.incoterm = order.type_id.incoterm_id.id
 
     @api.model
     def create(self, vals):
         if vals.get('name', '/') == '/'and vals.get('type_id'):
             type = self.env['sale.order.type'].browse(vals['type_id'])
```

Comment:
It should be better don't use built-in function name for your variable. Please rename it to `sale_type`.